### PR TITLE
Fix Devour Magic

### LINKED ISSUE #67: Felhunter Devour Magic
Felhunter dispells only friendly target on enemy it doesnt work

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/36902228-felhunter-devour-magic?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by justdothings**
**Issues addressed:** Closes #67 

**Tests performed:** compiled on ubuntu 14.04 without errors, Tested ingame.

### CODE REVIEW

File: src/game/Spells/SpellScript.h [hunk 1]
```diff
@@ -130,6 +130,7 @@ enum SpellScriptHookType
     SPELL_SCRIPT_HOOK_EFFECT_LAUNCH_TARGET,
     SPELL_SCRIPT_HOOK_EFFECT_HIT,
     SPELL_SCRIPT_HOOK_EFFECT_HIT_TARGET,
+	SPELL_SCRIPT_HOOK_EFFECT_SUCCESSFUL_DISPEL,
```
Comment (ShinDarth) [inline src/game/Spells/SpellScript.h]: code indentation

File: src/game/Spells/SpellEffects.cpp [hunk 2]
```diff
@@ -2608,17 +2608,7 @@ void Spell::EffectDispel(SpellEffIndex effIndex)
...
-    if (m_spellInfo->SpellFamilyName == SPELLFAMILY_WARLOCK && m_spellInfo->GetCategory() == SPELLCATEGORY_DEVOUR_MAGIC)
-    {
-        int32 heal_amount = m_spellInfo->Effects[EFFECT_1].CalcValue();
-        m_caster->CastCustomSpell(m_caster, 19658, &heal_amount, NULL, NULL, true);
-        // Glyph of Felhunter
-        if (Unit* owner = m_caster->GetOwner())
-            if (owner->GetAura(56249))
-                owner->CastCustomSpell(owner, 19658, &heal_amount, NULL, NULL, true);
-    }
+   CallScriptSuccessfulDispel(effIndex);
```
Comment (ShinDarth) [inline src/game/Spells/SpellEffects.cpp]: code indentation

File: src/game/Spells/SpellScript.h [hunk 3]
```diff
@@ -292,6 +293,7 @@ class SpellScript : public _SpellScript
         HookList<EffectHandler> OnEffectLaunchTarget;
         HookList<EffectHandler> OnEffectHit;
         HookList<EffectHandler> OnEffectHitTarget;
+		HookList<EffectHandler> OnEffectSuccessfulDispel;
```
Comment (ShinDarth) [inline src/game/Spells/SpellScript.h]: code indentation

File: src/game/Spells/SpellScript.h [hunk 4]
```diff
@@ -333,6 +335,8 @@ class SpellScript : public _SpellScript
         // 10. OnEffectHitTarget - executed just before specified effect handler call - called for each target from spell target map
         // 11. OnHit - executed just before spell deals damage and procs auras - when spell hits target - called for each target from spell target map
         // 12. AfterHit - executed just after spell finishes all it's jobs for target - called for each target from spell target map
+		// this hook is only executed after a successful dispel of any aura
```
Comment (ShinDarth) [inline src/game/Spells/SpellScript.h]: code indentation

File: src/game/Spells/SpellScript.cpp [hunk 5]
```diff
@@ -310,6 +310,10 @@ bool SpellScript::_Validate(SpellInfo const* entry)
         if (!(*itr).GetAffectedEffectsMask(entry))
             sLog->outError("TSCR: Spell `%u` Effect `%s` of script `%s` did not match dbc effect data - handler bound to hook `OnEffectHitTarget` of SpellScript won't be executed", entry->Id, (*itr).ToString().c_str(), m_scriptName->c_str());
 
+	for (std::list<EffectHandler>::iterator itr = OnEffectSuccessfulDispel.begin(); itr != OnEffectSuccessfulDispel.end(); ++itr)
+        if (!(*itr).GetAffectedEffectsMask(entry))
+            sLog->outError("scripts", "Spell `%u` Effect `%s` of script `%s` did not match dbc effect data - handler bound to hook `OnEffectSuccessfulDispel` of SpellScript won't be executed", entry->Id, (*itr).ToString().c_str(), m_scriptName->c_str());
+
```
Comment (ShinDarth) [inline src/game/Spells/SpellScript.cpp]: code indentation

**Comment by ShinDarth** [comment]
@justdothings thanks for this PR, can you fix all code indentation problems that I pointed? set tab = 4 spaces in your editor

thanks

**Comment by DevRival** [comment]
What is it? As if this is not the complete code. You just cut out the part about this and removed the spell work itself.

**Comment by justdothings** [comment]
@DevRival no it works well im using on my server, 50+ online

**Comment by DevRival** [comment]
You see the code? What have you done? You have removed the spell work and made ​​only a new hook.

**Comment by justdothings** [comment]
@DevRival come on my server and test it :) you can close pr if you don't like.

**Comment by mik1893** [comment]
this is nonsense as @DevRival said...

**Pull request closed without merging**